### PR TITLE
Add page for transaction de-duplication

### PR DESCRIPTION
**Pull request by lachlanjc**
- Adds page at /transactions/dedupe
- Picks random transactions of all 3 types & decides randomly whether they have canonical transactions—that logic needs to be implemented
- Not sure how we want to wire up the merge/separate actions, so left as buttons for now
- Dynamic modal to show JSON details of every transaction

Screencast on Slack

### CODE REVIEW

File: app/views/admin/transaction_dedupe.html.erb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  <div class="card__banner flex justify-center items-center flex-wrap">
+    <div class="btn-group">
+      <button class="btn">
+        <%= inline_icon 'expand' %>
+        Separate each
+      </button>
+      <button class="btn bg-accent">
+        <%= inline_icon 'enter' %>
+        Merge into one
+      </button
```
Comment (motdotla) [inline app/views/admin/transaction_dedupe.html.erb:54]: nice. both of these we can turn into a couple services with a couple endpoints backing them. we laid the foundation for that last week on the back-end.

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -84,6 +84,27 @@ def negative_events
...
+      OpenStruct.new(
+        event: hq,
+        txs: hq.stripe_authorizations.limit(3)
+      ),
+      OpenStruct.new(
+        event: penn,
+        txs: penn.emburse_transactions.limit(3)
+      )
+    ]
+  end
```
Comment (motdotla) [inline app/controllers/admin_controller.rb:89]: Max has this data available now as well - https://github.com/hackclub/bank/pull/984 (just needs I think one final change and then can probably merge)

**Review by motdotla — COMMENTED** [review]
looks sharp. left comments related to what Max and I need to provide.

**Comment by motdotla** [comment]
@maxwofford let's get https://github.com/hackclub/bank/pull/984/files merged to master today - time allowing. That then allows Lachlan to take the final step with this PR - pulling in real data.

**Comment by lachlanjc** [comment]
Sorry, based on scanning through that PR I'm not positive how to use it. Just `HashedTransaction.marked_duplicate.group_by(&:event)`?

**Comment by maxwofford** [comment]
@lachlanjc does an array of duplicate TXs like this work? `TransactionEngine::HashedTransactionService::GroupedDuplicates.new.run`

**Review by motdotla — APPROVED** [review]
This looks good at this point. Reaches our goal to put this in front of ops sooner than later. We can come back and iterate further on it as necessary later. :+1:

**Comment by motdotla** [comment]
I'll be merging this soon - likely tomorrow. Thanks for this @lachlanjc

**Comment by motdotla** [comment]
more likely merging tomorrow or Monday.

instead, I've been pushing the emburse historical matching forward to get some baselines on how the data is coming together vs current production data (across some low-transaction events to begin)

**Pull request merged**